plugins/LadspaEffect/calf/src/plugin.cpp

```diff
@@ -467,14 +469,17 @@ ladspa_plugin_metadata_set::~ladspa_plugin_metadata_set()
     delete []descriptor.PortNames;
     delete []descriptor.PortDescriptors;
     delete []descriptor.PortRangeHints;
+    free(descriptorName);
 #if USE_DSSI
     if (presets)
         presets->clear();
     if (preset_descs)
         preset_descs->clear();
     delete presets;
     delete preset_descs;
+    free(descriptorForDssiName);
 #endif
+    delete metadata;
```

Comment:
`metadata` is `const`. You should go to the point where `metadata` is created, make sure you keep a reference, and release it on destruction; you may need a function with `__attribute__((destructor))`.